src/main/java/com/puppycrawl/tools/checkstyle/checks/design/VisibilityModifierCheck.java

```diff
@@ -825,7 +822,7 @@ private boolean isImmutableField(DetailAST variableDef) {
         if (isFinalField(variableDef)) {
             final DetailAST type = variableDef.findFirstToken(TokenTypes.TYPE);
             final boolean isCanonicalName = isCanonicalName(type);
-            final String typeName = getTypeName(type, isCanonicalName);
+            final String typeName = getCanonicalName(type);
```

Comment:
There is a big difference here, so this should be killable.
Example: 
Code: `ImmutableSet<String>`
Results: `ImmutableSet` vs `String`

Reply:
My original analysis was on a clean master. It turns out this is now a cascade from https://github.com/checkstyle/checkstyle/pull/12643#discussion_r1070767748 .

When we removed the `getFirstChild` in `getCanonicalName`, it became the exact same as `type.getFirstChild().getText()`.

The only time it is not exactly the same is on primitive types as they are not `IDENT`s but `LITERAL_...`. 

Unfortunately it doesn't matter with the if/else after the call in `isImmutableField` because the other checks are working. If we say "int" is immutable, it has no type arguments and returns true for the if. Primitives are not canonical and `isPrimitive` will identify the primitive and returns true for the else.

The other usage is in `getTypeArgsClassNames`. The problem is primitives can't be a type argument (IE: Class<T>). So I am not seeing a way for this to be killed.